packages/wrangler/src/api/startDevWorker/LocalRuntimeController.ts

```diff
@@ -48,118 +48,20 @@ function getName(config: StartDevWorkerOptions) {
-		} else if (binding.type === "data_blob") {
-			bindings.data_blobs ??= {};
-			bindings.data_blobs[name] = await getBinaryFileContents(binding.source);
-		} else if (binding.type === "browser") {
-			bindings.browser = { binding: name };
-		} else if (binding.type === "ai") {
-			bindings.ai = { binding: name };
-		} else if (binding.type === "version_metadata") {
-			bindings.version_metadata = { binding: name };
-		} else if (binding.type === "durable_object_namespace") {
-			bindings.durable_objects ??= { bindings: [] };
-			bindings.durable_objects.bindings.push({ ...binding, name: name });
-		} else if (binding.type === "queue") {
-			bindings.queues ??= [];
-			bindings.queues.push({ ...binding, binding: name });
-		} else if (binding.type === "r2_bucket") {
-			bindings.r2_buckets ??= [];
-			bindings.r2_buckets.push({ ...binding, binding: name });
-		} else if (binding.type === "d1") {
-			bindings.d1_databases ??= [];
-			bindings.d1_databases.push({ ...binding, binding: name });
-		} else if (binding.type === "vectorize") {
-			bindings.vectorize ??= [];
-			bindings.vectorize.push({ ...binding, binding: name });
-		} else if (binding.type === "constellation") {
-			bindings.constellation ??= [];
-			bindings.constellation.push({ ...binding, binding: name });
-		} else if (binding.type === "hyperdrive") {
-			bindings.hyperdrive ??= [];
-			bindings.hyperdrive.push({ ...binding, binding: name });
-		} else if (binding.type === "service") {
-			bindings.services ??= [];
-			bindings.services.push({ ...binding, binding: name });
-		} else if (binding.type === "fetcher") {
-			fetchers[name] = binding.fetcher;
-		} else if (binding.type === "analytics_engine") {
-			bindings.analytics_engine_datasets ??= [];
-			bindings.analytics_engine_datasets.push({ ...binding, binding: name });
-		} else if (binding.type === "dispatch_namespace") {
-			bindings.dispatch_namespaces ??= [];
-			bindings.dispatch_namespaces.push({ ...binding, binding: name });
-		} else if (binding.type === "mtls_certificate") {
-			bindings.mtls_certificates ??= [];
-			bindings.mtls_certificates.push({ ...binding, binding: name });
-		} else if (binding.type === "logfwdr") {
-			bindings.logfwdr ??= { bindings: [] };
-			bindings.logfwdr.bindings.push({ ...binding, name: name });
-		} else if (binding.type.startsWith("unsafe-")) {
-			bindings.unsafe ??= {
-				bindings: [],
-				metadata: undefined,
-				capnp: undefined,
-			};
-			bindings.unsafe.bindings?.push({ ...binding, name: name });
-		}
-	}
+	// TODO: Remove this passthrough
+	const bindings = event.config._bindings
+		? event.config._bindings
+		: convertedBindings;
 
-	const persistence = event.config.dev?.persist
-		? getLocalPersistencePath(
-				typeof event.config.dev?.persist === "object"
-					? event.config.dev?.persist.path
-					: undefined,
-				event.config.config?.path
-			)
-		: null;
+	const persistence = getLocalPersistencePath(
```

Comment:
what changed that makes the `event.config.dev?.persist` check irrelevant here? Since `persist` is optional, won't `event.config.dev?.persist.path` potentially break?

Reply:
This matches existing behaviour (the diff is because I'd previously misunderstood how this should work). See https://github.com/cloudflare/workers-sdk/blob/7d93a60deeb8b64ca8ff8be66860ffbda4b860ad/packages/wrangler/src/dev.tsx#L848-L851.

Essentially, `getLocalPersistencePath()` treats the first argument as `string | undefined`

Reply:
Let's let typescript checks handle these kind of nits – fair comment if the typescript checks fail but in this case they are passing

Reply:
`persist` is optional, but `path` is required within `persist`, and so we don't need an extra `?`